virttest/libvirt_xml/devices/interface.py

```diff
@@ -232,3 +251,53 @@ def __init__(self, virsh_instance=base.base.virsh):
                                      tag_name="guest")
             super(self.__class__, self).__init__(virsh_instance=virsh_instance)
             self.xml = '<driver/>'
+
+    class Vlan(base.base.LibvirtXMLBase):
+
+        """
+        Interface vlan xml class.
+
+        Properties:
+
+        vlan:
+            attribute.
+        tags:
+            list. tags element dict list
+         """
+        __slots__ = ("vlan_trunk", "tags")
+
+        def __init__(self, virsh_instance=base.base.virsh):
+            accessors.XMLAttribute(property_name="vlan_trunk",
+                                   libvirtxml=self,
+                                   forbidden=None,
+                                   parent_xpath='/',
+                                   tag_name='vlan',
+                                   attribute='trunk')
+            accessors.XMLElementList(property_name='tags',
+                                     libvirtxml=self,
+                                     parent_xpath='/',
+                                     marshal_from=self.marshal_from_tag,
+                                     marshal_to=self.marshal_to_tag)
+            super(self.__class__, self).__init__(virsh_instance=virsh_instance)
+            self.xml = '<vlan/>'
+
+        @staticmethod
+        def marshal_from_tag(item, index, libvirtxml):
+            """Convert a dictionary into a tag + attributes"""
+            del index  # not used
+            del libvirtxml  # not used
+            if not isinstance(item, dict):
+                raise xcepts.LibvirtXMLError("Expected a dictionary of tag "
+                                             "attributes, not a %s"
+                                             % str(item))
+                # return copy of dict, not reference
+            return ('tag', dict(item))
```

Comment:
I was going to suggest to remove redundant parentheses, but it seems everyone is keeping them, then it's ok to me.